### DISCUSSION TITLE
Is it possible to support the installation method of npm install --omit=optional?

### DISCUSSION THREAD

**Discussion by filefoxper** [web-infra-dev/rspack#8764] [Ideas]
在一些大型项目中，有大量只服务于开发过程的包，通常会被开发团队放在 optionalDependencies 中。在生产环境使用 npm install --omit=optional 的安装方式可以忽略这些只服务于开发过程的包。而 npm install --omit 并没有提供针对个别包的豁免权，这会导致 @rspack/binding 无法正常安装核心包的问题（因为核心包是放在 @rspack/binding/package.json 的 optionalDependencies 中的）。

思路：可以学习一下 esbuild ，在@rspack/binding 包的 package.json 中加入 postinstall 脚本，在安装完毕时，检查并补全缺失的包。希望可以支持一下，谢谢。

**Comment by chenjiahan** [comment]
See https://github.com/web-infra-dev/rspack/issues/8762
